flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/source/IcebergSource.java

```diff
@@ -225,10 +254,12 @@ public static class Builder<T> {
     private SplitAssignerFactory splitAssignerFactory;
     private SerializableComparator<IcebergSourceSplit> splitComparator;
     private ReaderFunction<T> readerFunction;
+    private TypeInformation<T> outputTypeInfo;
```

Comment:
Ideally, `TypeInformation` should be part of `ReaderFunction` interface. but unfortunately, `ReaderFunction` is right now defined as a `@FunctionalInterface`. Not sure changing it and adding a new method would break compatibility.